SU2_CFD/src/fluid/CMutationTCLib.cpp

```diff
@@ -27,46 +27,201 @@
 
 #include "../../include/fluid/CMutationTCLib.hpp"
 
+#ifdef HAVE_MPP
+
 CMutationTCLib::CMutationTCLib(const CConfig* config, unsigned short val_nDim): CNEMOGas(config, val_nDim){
+ 
+  Mutation::MixtureOptions opt(gas_model);
+  string transport_model, state_model;	
+
+  /* Allocating memory*/
+  Cv_ks.resize(nEnergyEq*nSpecies,0.0);
+  h_RT.resize(nSpecies,0.0);
+  es.resize(nEnergyEq*nSpecies,0.0);
+  omega_vec.resize(1,0.0);
+
+  /*--- Set up inputs to define type of mixture in the Mutation++ library ---*/
+
+  /*--- Define thermochemical nonequilibrium model ---*/
+  if(!frozen)
+    state_model = "ChemNonEqTTv";
+  else 
+    state_model = "NonEqTTv";
+
+  /*--- Define transport model ---*/
+  if(Kind_TransCoeffModel == WILKE)
+    transport_model = "Wilke";
+  else if (Kind_TransCoeffModel == GUPTAYOS)
+ 	transport_model = "Gupta-Yos";
+  
+  opt.setStateModel(state_model);
+  opt.setViscosityAlgorithm(transport_model);
+  opt.setThermalConductivityAlgorithm(transport_model); 
+  
+  /* Initialize mixture object */
+  mix.reset(new Mutation::Mixture(opt));
+
+  for(iSpecies = 0; iSpecies < nSpecies; iSpecies++) MolarMass[iSpecies] = 1000* mix->speciesMw(iSpecies); // x1000 to have Molar Mass in kg/kmol
+
+  if (mix->hasElectrons()) { nHeavy = nSpecies-1; nEl = 1; }
+  else                     { nHeavy = nSpecies;   nEl = 0; }
+
+}
+
+CMutationTCLib::~CMutationTCLib(){}
+  
+void CMutationTCLib::SetTDStateRhosTTv(vector<su2double>& val_rhos, su2double val_temperature, su2double val_temperature_ve){
+
+  temperatures[0] = val_temperature;
+  temperatures[1] = val_temperature_ve; 
+
+  T   = temperatures[0];
+  Tve = temperatures[1];
 
-  //CGarbacz: if wilke - transportmodel = 'wilke' and so on;
+  rhos = val_rhos;
 
-  //CGarbacz: nEl = mix.getnumberelectrons; nHeavy = nSpecies-nEl;
+  Density = 0.0;
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+    Density += rhos[iSpecies];
 
-  //CGarbacz: if frozen send nonchecmttv
+  Pressure = GetPressure();
```

Comment:
Does "Get"Pressure still exist?  Or is it Compute now?

Reply:
yup you're right. the compiler did not complain... weird...

Reply:
it is actually a typo. but interestingly the compiler does not complain since CFluidModel has a

su2double GetPressure() const { return Pressure; }

but here we really want to compute it, so I will fix it, thanks